q2_sample_classifier/classify.py

```diff
@@ -38,6 +41,47 @@
 }
 
 
+def classify_samples_from_dist(ctx, dmtx, metadata, k,
+                               palette=defaults['palette']):
+    ''' Returns knn classifier results from a distance matrix.'''
+    distance_matrix = dmtx.view(skbio.DistanceMatrix)
+    predictions = []
+    metadata_series = metadata.to_series()
+    for i, row in enumerate(distance_matrix):
```

Comment:
it looks like you can just use [condensed_form](http://scikit-bio.org/docs/0.5.1/generated/generated/skbio.stats.distance.DistanceMatrix.condensed_form.html#skbio.stats.distance.DistanceMatrix.condensed_form) to replicate this operation (then use `reindex` and `dropna` to resort and subset the metadata categories). Right?

Reply:
Don't immediately see how, though I admit this is a pretty dumb brute force approach. condensed_form on the distance_matrix would require that we keep track of which pairs each value belongs to, yes?

Might be a little while until I can look at this more, so feel free to edit if you like.

Reply:
I think you're right about needing to keep track of pairs... but I have not given it a try. Seems like we should be able to get that info from the raw indices.

Nothing wrong with brute force... just wondering if there's a more efficient approach.